app/models/conference/importer.rb

```diff
@@ -0,0 +1,73 @@
+require 'csv'
+class Conference::Importer
+  
+  ## This importer is depending on agreed-upon input format.
+  ## Input file:
+  # - Dates should be formatted with dd mm yyyy
+  # - UID should be unique and formatted with 'current season' + id: 2017001
+  # - The headers should not be changed
+  # Output:
+  # Conferences will be updated or created
+  # Conferences deleted in the input file, are not removed from the table
+  # Import errors are logged in a separate 'imports_logger' file
+  
+  class << self
+    def import(file)
+      check_valid(file)
+      info_log(:started, file)
+      process_csv(file)
+      info_log(:finished, file)
+    end
+    
+    private
+    
+    def info_log(arg, file)
+      notices = {
+        started: "Started importing file #{file.original_filename}",
+        finished: "Finished updating/creating #{count_conferences_in(file)} conferences"
+      }
+      imports_logger.info "\n***" + notices[arg] + "***"
+    end
+   
+    def imports_logger
+      Logger.new("#{Rails.root}/log/imports_logger.log")
+    end
+    
+    def check_valid(file)
+      raise "Oops! I can upload .csv only :-(" unless file.content_type == "text/csv"
+    end
+    
+    def count_conferences_in(file)
+      CSV.foreach(file.path, headers: true).count
+    end
+    
+    def process_csv(file)
+      CSV.foreach(file.path, headers: true, col_sep: ';' ) do |row|
+        begin
+          conference = Conference.find_or_initialize_by(gid: row['UID'])
+          # TODO check: when extra columns are added in the Google Doc, what happens?
+          conference_hash = {
+            gid: row['UID'].to_i,
```

Comment:
Because I don't like to make promises I can't keep. There is no way we can be sure of the uniqueness of the ID in the imported Google Doc. If it is called `uid`, we suggest that we can trust the uniqueness.